pytest_socket.py

```diff
@@ -3,62 +3,92 @@
 
 import pytest
 
-_true_socket = socket.socket
+_true_connect = socket.socket.connect
 
 
 class SocketBlockedError(RuntimeError):
-    def __init__(self, *args, **kwargs):
-        super(SocketBlockedError, self).__init__("A test tried to use socket.socket.")
+    def __init__(self, allowed, host, *args, **kwargs):
+        if allowed:
+            allowed = ','.join(allowed)
+        super(SocketBlockedError, self).__init__(
+            'A test tried to use socket.socket.connect() with host "{0}" (allowed: "{1}").'.format(host, allowed)
```

Comment:
What value does reporting the allowed hosts in this error message add?

Reply:
It's just takes the mental work out of checking which hosts you allowed against which got blocked. Especially if you have allowed hosts at different levels (e.g. via cli *and* marked). Especially good if it's a typo and you want to see both values near each other.